lib/Solver/CachingSolver.cpp

```diff
@@ -33,12 +33,17 @@ class CachingSolver : public SolverImpl {
   
   struct CacheEntry {
     CacheEntry(const ConstraintSet &c, ref<Expr> q)
-      : constraints(c), query(q) {}
+        : constraints(c.begin(), c.end()), query(q) {
+      // Order constraints
+      std::sort(
```

Comment:
I see the benefit, but what about the impact on performance?

Reply:
That's hard to estimate. For every lookup the constraints are sorted.
We can remove this commit for now - and add this optimisation later or make sure it's always sorted in the first place,
or we add an option. Those options are all fine for me.